src/metabase/models/notification.clj

```diff
@@ -370,6 +372,71 @@
           :send_once      false}
          instance))
 
+;; ------------------------------------------------------------------------------------------------;;
+;;                                         Permissions                                             ;;
+;; ------------------------------------------------------------------------------------------------;;
+
+(defn current-user-can-read-payload?
+  "Check if the current user can read the payload of a notification."
+  [notification]
+  (case (:payload_type notification)
+    :notification/card
+    (mi/can-read? :model/Card (-> notification :payload :card_id))
+
+    :notification/system-event
+    (mi/superuser?)
+
+    :notification/testing
+    true))
+
+(defn current-user-is-recipient?
+  "Check if the current user is a recipient of a notification."
+  [notification]
+  (->> (:handlers (t2/hydrate notification [:handlers :recipients]))
+       (mapcat :recipients)
+       (map :user_id)
+       set
+       (some #{(mi/current-user-id)})
+       boolean))
+
+(defn current-user-is-creator?
+  "Check if the current user is the creator of a notification."
+  [notification]
+  (= (:creator_id notification) (mi/current-user-id)))
+
+(defmethod mi/can-read? :model/Notification
+  ([notification]
+   ;; users can view a notification if they can view the payload or if they are the creator
+   (or
+    (mi/superuser?)
+    (current-user-is-creator? notification)
+    (current-user-is-recipient? notification)))
+  ([_ pk]
+   (mi/can-read? (t2/select-one :model/Notification pk))))
+
+(defmethod mi/can-create? :model/Notification
+  [_ notification]
+  (or
+   (mi/superuser?)
+   (current-user-can-read-payload? notification)
+   ;; if ee is enabled, prevent users without subscription permissions from creating notifications
+   (and
+    (premium-features/has-feature? :advanced-permissions)
+    (perms/current-user-has-application-permissions? :subscription))))
+
+(defmethod mi/can-update? :model/Notification
+  [instance _changes]
+  (or
+   (mi/superuser?)
+   (and
+    (current-user-is-creator? instance)
+    ;; if advanced-permissions is enabled, we require users to have subscription permissions
+    ;; and is the owner of the notification and can read the payload
```

Comment:
```suggestion
    ;; if advanced-permissions is enabled, we also require users to have subscription permissions
```

Reply:
It's strange that this logic is inconsistent with that for creating notifications - is this a bug?

Reply:
it's intentionally actually

see 3.b from Katya's [message](https://metaboat.slack.com/archives/C07318NQ5M2/p1734630525977029?thread_ts=1734596426.664419&cid=C07318NQ5M2)

Reply:
tagging @KorowkaMu in to make sure I parsed it correctly.

So in case advanced permissions is enable, non-admin users can only edit notification if:
- they are the creator
- they can view the card
- they have subscriptions permissions

this is not related to unsubscribe permissions, a recipient can always unsubscribe themselves from any notifications.